monitor/events.go

```diff
@@ -3,76 +3,263 @@ package monitor
 import (
 	"bytes"
 	"encoding/json"
-	"fmt"
 	"net/http"
+	"strconv"
+	"time"
 
 	"github.com/golang/glog"
+	ethTypes "github.com/livepeer/go-livepeer/eth/types"
 )
 
-type Event struct {
-	Name       string            `json:"event"`
-	Properties map[string]string `json:"properties"`
+var Enabled bool
+
+type metricsVars struct {
+	lastSegmentEmergedAt time.Time
+	lastSegmentNonce     uint64
+	lastSeqNo            int64
+	segmentsInFlight     int
 }
 
-var EventsURL = "http://metrics.livepeer.org/events"
+var metrics = metricsVars{}
+
+type event struct {
+	Name       string                 `json:"event"`
+	Nonce      string                 `json:"nonce"`
+	Properties map[string]interface{} `json:"properties"`
+}
+
+var eventsURLBase string
+var eventsURL string
+
+var network = "rinkeby"
+
+func SetNetwork(n string) {
+	network = n
+	eventsURL = eventsURLBase + n + "/events"
```

Comment:
Yes, we should enforce url structure, either in events.go or in livepeer.go - or else someone could easily run node on rinkeby and send metrics to the /mainnet endpoint. Or else we should send `network` field with each event

Reply:
Yeah that's possible, but we're also banking a lot on trust with the metrics system, since anyone can modify anything. We can hard-code "http://metrics.livepeer.org/v2/rinkeby/events" or "http://metrics.livepeer.org/v2/mainnet/events" as the monhost. See how it's done for Infura.

https://github.com/livepeer/go-livepeer/blob/2e9b5dacb961d2027d189c7603bdf07220dce121/cmd/livepeer/livepeer.go#L107-L117

Reply:
My point is that when we are not enforcing url scheme we are unnecessarily creating possibility for errors in configuration - users will not to remember that two commandline arguments should be change together.
Also, same scheme is enforced on receiving side - so, if user will want to have receiving url not to be `xxx/(mainnet|rinkeby)/events` - he will need to change source code of metrics server anyway, and if he go thus far, he will change goclient's code without problem

Reply:
They shouldn't have to set the URL themselves as long as `-monitor` is true. I can see `/events` being hardcoded on the metrics server, but I don't think the network name should be.